### DISCUSSION TITLE
Editing fails.json

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#820] [Q&A]
I wanted to modify the fail text when the game recommends a scrub, to clarify that the rest of the mission will suffer a penalty regardless.  I have two questions about fails.json.

1. Do you get the button to scrub the mission in each scenario where it says engineers, ground control, or mission control recommends a scrub?
2. Am I right in thinking that the "text" section of the .json file can be of arbitrary length (up to a point)?  So it's not a set length like it was in the old binary file and it can be edited without causing trouble?

![Update the help file, saying all steps going forward will have penalty regardless whether you push button](https://github.com/raceintospace/raceintospace/assets/3118014/7ed91c71-e016-4544-83e6-513afb3c1db6)

**Comment by hweimer** [comment]
1. The button appears when the code of the failure is set to 15, irrespective of what the text says.
2. The text can be of arbitrary length, as long as it is shorter than 200 characters (I doubt that more would fit into the text box anyway).

**Reply by peyre** [reply]
Thanks!

That can't be _quite_ right though, I think: the first in the list, for instance, is a 15 and the text says "MAJOR FIRE ON THE PAD. LAUNCH IS CANCELED. ROCKET AND PAYLOAD DESTROYED."  But I guess any 15 where a scrub is recommended must be one that will show the button.

In fact, a lot of 15s just say the mission is scrubbed, and the example above shows a "fail" value of 97.

**Reply by hweimer** [reply]
The MAJOR FIRE ON PAD failure has a *fail* value of 15, but a *code* value of 2 (the latter signalling an end-of-mission failure). The fail value controls which video sequence is being played (i.e., MissionIdSequences ending with "R015" in fails.json). A fail of 97 usually refers to a video sequence showing one of these blinking rings (not sure what that actually is, I think it is the same as the "temporary loss of signal" thing in the floppy version).

**Reply by peyre** [reply]
Wait, are you sure of that?  I'm probably misunderstanding somewhere, but what I see is:

  {
    "MissionStep": "F009",
    "percentage": 5100,
    "code": 15,
    "val":  0,
    "xtra":  0,
    "fail": 97,
    "text": "IGNITION FAILURE. REPAIRS ARE MADE AND A SUBSTITUTE THRUST CAN MAKE THE BURN. GROUND CONTROL RECOMMENDS TO SCRUB MISSION."
  },
...
  {
    "MissionStep": "F010",
    "percentage": 5100,
    "code": 15,
    "val":  0,
    "xtra":  0,
    "fail": 97,
    "text": "IGNITION FAILURE. REPAIRS ARE MADE AND A SUBSTITUTE THRUST CAN MAKE THE BURN. GROUND CONTROL RECOMMENDS TO SCRUB MISSION."
  },
...
  {
    "MissionStep": "F011",
    "percentage": 5100,
    "code": 15,
    "val":  0,
    "xtra":  0,
    "fail": 97,
    "text": "IGNITION FAILURE. REPAIRS ARE MADE AND A SUBSTITUTE THRUST CAN MAKE THE BURN. GROUND CONTROL RECOMMENDS TO SCRUB MISSION."
  },

**Reply by hweimer** [reply]
Yes, I am sure. I also don't see how any of these failures contradict what I said. All of them are code 15 (i.e., downgrade plus option to scrub) and have a fail value of 97, which leads to a video showing something like that (ring_06.ogg):
![Screenshot from 2023-11-21 19-54-28](https://github.com/raceintospace/raceintospace/assets/68769393/e81b1711-66e3-4ace-8715-69c062c862c8)

I guess it is somewhat confusing that the data structure contains both a "code" entry (which controls the game logic) and a "fail" entry (which is only relevant for selecting the appropriate video sequence). Maybe the latter should have been called "fseq" or something, so that it is clear that it has something to do with the sequence data in fseq.json.

Does this somehow make sense?

**Reply by peyre** [reply]
I had a feeling that might have been what was tripping me up.  Thanks!
